core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/service/ClusterProxyService.java

```diff
@@ -136,15 +136,17 @@ private ConfigRegistrationRequest createProxyConfigReRegisterRequest(Stack stack
         ConfigRegistrationRequestBuilder requestBuilder = new ConfigRegistrationRequestBuilder(stack.getResourceCrn())
                 .withAliases(singletonList(clusterId(stack.getCluster())))
                 .withServices(serviceConfigs(stack))
```

Comment:
Just a sanity-check: Is it really alright to keep using a common set of service configs (and have distinction with respect to CCM version only for the Knox URL)? What I mean is the removal of `serviceConfigsForNoCcmAndCcmV1()` that happened in the preceding PR https://github.com/hortonworks/cloudbreak/pull/11597/files#diff-d7b11aa8082eb03585c7256fe16537e78ab246a09d789ac3a376cac0e6b0d743 , when that method got mostly replaced with `serviceConfigsForCcmV2()` (renamed to simply `serviceConfigs()`).

Reply:
@lajosrodek 
Yes, I tested with IDBroker sync again and autoscaling, because it turned out that component uses CP to reach Yarn through Knox, which failed in the mentioned ticket.
When the serviceConfigs has been simplified/merged we moved away from using `/clouderamanager/` CM backdoor for reaching CM APIs through CP and CCMv1. WIth this PR it remains the same, but the merge of Knox url generation was a bad choice as CCMv2 approaches has been built on top of that limitation that NGiNX doesn't verify the client certificate. Which means this behaviour makes it possible to work Knox through CP with CCMv2 approaches and not with direct ClusterProxy and/or CCMv1 as the client cert validation is enabled for those scenarios.